proposals/geolocation.md

```diff
@@ -0,0 +1,404 @@
+  "links": [
+     {
+        "href": "./properties/location",
+        "rel": "g:geolocationOf"
+     }
+  ],
+   ...
+  "properties": {
+    "location": {
+      "title": "Get current location",
+      "description": "Use sensor fusion to estimate location, orientation, and motion",
+      "observable": true,
+      "readOnly": true,
+      "type": "object",
+      "properties": {
+         "position": {
+            "longitude": {
+                "type": number,
+                "unit": "qudt:Angle"
+            },
+            "latitude": {
+                "type": number,
+                "unit": "qudt:Angle"
+            },
+            "accuracy": {
+                "type": number
+            },
+            "accuracyUnit": {
+                "type": "qudt:Length"
+            }
+         },
+         "altitude": {
+            "elevation": {
+                "type": number,
+                "unit": "qudt:Length"
+            },
+            "elevationUnit": {
+                "type": "qudt:Length"
+            },
+            "depth": {
+                "type": number,
+                "unit": "qudt:Length"
+            },
+            "depthUnit": {
+                "type": "qudt:Length"
+            },
+            "accuracy": {
+                "type": number,
+                "unit": "qudt:Length"
+            },
+            "accuracyUnit": {
+                "type": "qudt:Length"
+            }
+         },
+         "velocity": {
+            "x": {
+                "type": number,
+                "type": "qudt:Speed"
+            },
+            "y": {
+                "type": number,
+                "type": "qudt:Speed"
+            },
+            "z": {
+                "type": number,
+                "type": "qudt:Speed"
+            },
+            "velocityUnit": {
+                "type": "qudt:Speed"
+            },
+            "accuracy": {
+                "type": number,
+                "type": "qudt:Speed"
+            },
+            "accuracyUnit": {
+                "type": "qudt:Speed"
+            }
+         },
+         "acceleration": {
+            "x": {
+                "type": number,
+                "type": "qudt:Acceleration"
+            },
+            "y": {
+                "type": number,
+                "type": "qudt:Acceleration"
+            },
+            "z": {
+                "type": number,
+                "type": "qudt:Acceleration"
+            },
+            "accelerationUnit": {
+                "type": "qudt:Acceleration"
+            },
+            "accuracy": {
+                "type": number,
+                "type": "qudt:Acceleration"
+            },
+            "accuracyUnit": {
+                "type": "qudt:Acceleration"
+            }
+         },
+      },
+      "forms": [
+        {
+          "href": "http://geolocator.thing/api/location",
+          "contentType": "application/json"
+        }
+      ]
+    }
+  }
+}
+
+```
+
+#### External Geolocation Device
```

Comment:
The Geolocation can also be provided by a service which the Thing will send a request to (e.g., send an image and, based on visual localization service, the position and orientation are returned to the Thing.

Reply:
Oh, that's interesting.  Closely related to edge computing.  However, I think that we should put that under "extensions" for now.